.azure-pipelines/ultimate-pipeline.yml

```
@@ -3811,6 +3811,10 @@ stages:
 
     steps:
       - checkout: none
+
+      - script: sudo apt-get update && apt-get install python3.9-venv
```

AFAICT [Python is already cached on the runners](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#python).

If you switch to the built in python task, you might see some improvements in install speed
```suggestion
      - task: UsePythonVersion@0
        inputs:
            versionSpec: '3.9.x'
```